### DISCUSSION TITLE
playTone() with Telemetrix ?

### DISCUSSION THREAD

**Discussion by fguenego** [MrYsLab/telemetrix#45] [Q&A] [answered]
Hello,

I use the Telemetrix library after having noticed that the basic library "pyfirmata" did not allow me to use easily a HC-SR04 sensor.
I congratulate you for your work but I am surprised not to find a function allowing to play sound with a buzzer. Especially since this function existed in pymata4. (in the example playTone.py, there were the following instructions:
```
# set a pin's mode for tone operations
board.set_pin_mode_tone(TONE_PIN)
# specify pin, frequency and duration and play tone
board.play_tone(TONE_PIN, 1000, 500)
# specify pin and frequency and play continuously
board.play_tone_continuously(TONE_PIN, 2000)
# specify pin to turn pin off
board.play_tone_off(TONE_PIN)
```

All these methods have disappeared from the TELEMETRIX library. 
I couldn't play the music scale because the delays in microseconds don't "pass".
Is there a solution with the existing methods to play the music scale?

Here is my test:

```
import sys
import time

from telemetrix import telemetrix

BassTab0=[1911,1702,1516,1431,1275,1136,1012]  
BUZZER_PIN = 10

def soundD(index):
    for i in range(100):
        board.digital_write(BUZZER_PIN,1)
        time.sleep(BassTab[index]/1000000.0) 
        board.digital_write(BUZZER_PIN,0)
        time.sleep(BassTab[index]/1000000.0)

board = telemetrix.Telemetrix()

board.set_pin_mode_digital_output(BUZZER_PIN)
board.digital_write(BUZZER_PIN,0)

try:
    for i in range(len(BassTab0)):
        soundD(i)
        time.sleep(2)
```

**Comment by MrYsLab** [comment, marked as answer]
@fguenego Thanks for the question. The Tone library was not included in Telemetrix because the Arduino Uno has limited space, and I wanted to add many more features, such as accelstepper support and SPI. In addition, the Tone library can interfere with PWM operation because it uses Arduino hardware timers. 

I know this does not meet your needs, but if you need both HC-SR04 support and Tone Library support, you may wish to use [pymata4](https://github.com/MrYsLab/pymata4). I archived pymata4 because adding new features to Firmata became more and more difficult. The Telemetrix design made adding new features much simpler and more robust.

Hopefully, pymata4 will solve your issue. Please let me know if you have any further questions.
